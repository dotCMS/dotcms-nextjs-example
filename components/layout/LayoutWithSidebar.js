import Sidebar from './Sidebar';

const COLUMS_TOTAL = 12;

const SIDEBAR_SIZE_MAP = {
    small: 2,
    medium: 3,
    large: 4
};

const getSidebarLayoutColumns = (sidebar) => {
    const sidebarSize = SIDEBAR_SIZE_MAP[sidebar.width];

    if (sidebar.location === 'right') {
        return {
            first: COLUMS_TOTAL - sidebarSize,
            second: sidebarSize
        };
    }

    return {
        first: sidebarSize,
        second: COLUMS_TOTAL - sidebarSize
    };
};

const LayoutWithSidebar = ({ sidebar, children }) => {
    const columns = getSidebarLayoutColumns(sidebar);

    return (
        <>
            {/* <BootstrapRow>
                <BootstrapCol md={columns.first}>
                    {sidebar.location === 'left' ? <Sidebar {...sidebar} /> : children}
                </BootstrapCol>
                <BootstrapCol md={columns.second}>
                    {sidebar.location === 'right' ? <Sidebar {...sidebar} /> : children}
                </BootstrapCol>
            </BootstrapRow> */}
        </>
    );
};

export default LayoutWithSidebar;
