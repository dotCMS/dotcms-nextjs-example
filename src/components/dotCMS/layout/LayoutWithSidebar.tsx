const COLUMS_TOTAL = 12

const SIDEBAR_SIZE_MAP = {
  small: 2,
  medium: 3,
  large: 4,
}

const getSidebarLayoutColumns = (sidebar: Record<string, any>) => {
  // @ts-ignore we can index with any but we need better types
  const sidebarSize = SIDEBAR_SIZE_MAP[sidebar.width]

  if (sidebar.location === 'right') {
    return {
      first: COLUMS_TOTAL - sidebarSize,
      second: sidebarSize,
    }
  }

  return {
    first: sidebarSize,
    second: COLUMS_TOTAL - sidebarSize,
  }
}

export type LayoutWithSidebarProps = {
  sidebar: Record<string, any>
  children: React.ReactNode
}

export const LayoutWithSidebar = ({
  sidebar,
  children,
}: LayoutWithSidebarProps) => {
  const columns = getSidebarLayoutColumns(sidebar)

  return (
    <>
      {/* <BootstrapRow>
                <BootstrapCol md={columns.first}>
                    {sidebar.location === 'left' ? <Sidebar {...sidebar} /> : children}
                </BootstrapCol>
                <BootstrapCol md={columns.second}>
                    {sidebar.location === 'right' ? <Sidebar {...sidebar} /> : children}
                </BootstrapCol>
            </BootstrapRow> */}
    </>
  )
}

export default LayoutWithSidebar
