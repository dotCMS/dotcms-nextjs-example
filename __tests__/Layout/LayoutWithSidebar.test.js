import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import LayoutWithSidebar from '../../components/dotcms/layout/LayoutWithSidebar';

describe('<LayoutWithSidebar />', () => {
    it('should render correctly', () => {
        const wrapper = shallow(
            <LayoutWithSidebar sidebar={{ location: 'right', width: 'medium', containers: [{}, {}] }}>
                <p>Hola Mundo</p>
            </LayoutWithSidebar>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});
