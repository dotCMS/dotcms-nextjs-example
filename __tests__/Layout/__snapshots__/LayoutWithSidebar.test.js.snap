// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutWithSidebar /> should render correctly 1`] = `
<Row
  tag="div"
>
  <Col
    md={9}
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <p>
      Hola Mundo
    </p>
  </Col>
  <Col
    md={3}
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Sidebar
      containers={
        Array [
          Object {},
          Object {},
        ]
      }
      location="right"
      width="medium"
    />
  </Col>
</Row>
`;
